backend/testClient.spec.js

```diff
@@ -0,0 +1,175 @@
+
+const { typeDefs } = require("./typeDefs");
+const { resolvers } = require("./resolvers");
+
+const { ApolloServer, gql } = require('apollo-server');
+
+const { jwt } = require('jsonwebtoken');
+
+let token;
+
+describe("ServerTest", () => { 
+
+  const server = new ApolloServer({
+    typeDefs,
+    resolvers
+  });
+
+  const { query, mutate } = createTestClient(server);
+
+  it('test Login', async () => {
+    const res = await mutate({ mutation: LOGIN_USER,
+      variables: { username: "user1", password: "12345"}
+    });
+    token = res.data.loginUser.token;
+
+    let res2 = await query({ query: GET_TODOS, 
+      variables: {token: token}})
+  
+    expect(res2).toMatchObject({
+      "data": {
+        "todos": [
+          {
+            "title": "SDF - Task 3",
+          },
+          {
+            "title": "SDF - Task 2",
+          }
+        ]
+      }
+    })
+  });
+
+  it('get all todos', async () => {
+    const res = await query({ query: GET_TODOS, variables: {token: token}
+    });
+    
+    expect(res).toMatchObject(
+      {
+        "data": {
+          "todos": [
+            {
+              "title": "SDF - Task 3",
+            },
+            {
+              "title": "SDF - Task 2",
+            }
+          ]
+        }
+      }
+    );
+  });
+
+  it('add new todo', async() => {
+    const res = await mutate({ mutation: ADD_NEW_TODO,
+    variables: { title: "new todo title", token: token }
+    });
+   const res2 = await query({ query: GET_TODOS, variables: {token: token}});
+    
+    expect(res2).toMatchObject(
+      {
+        "data": {
+          "todos": [
+            {
+              "title": "SDF - Task 3",
+            },
+            {
+              "title": "SDF - Task 2",
+            },
+            {
+              "title": "new todo title"
+            }
+          ]
+        }
+      }
+    );
+    
+  });
+
+  it('delete todo', async() => {
+    const res = await mutate({ mutation: DELETE_TODO,
+    variables: { index: 2, token: token }
+    });
+
+   const res2 = await query({ query: GET_TODOS, variables: {token: token}});
+    
+    expect(res2).toMatchObject(
+      {
+        "data": {
+          "todos": [
+            {
+              "title": "SDF - Task 3",
+            },
+            {
+              "title": "SDF - Task 2",
+            }
+          ]
+        }
+      }
+    );
+    
+  });
+
+  it('update todo', async() => {
+    const res = await mutate({ mutation: UPDATE_TODO,
+    variables: {  title: "updated todo title", index: 1 , token: token}
+    });
+
+   const res2 = await query({ query: GET_TODOS, variables: {token: token}});
+    
+    expect(res2).toMatchObject(
+      {
+        "data": {
+          "todos": [
+            {
+              "title": "SDF - Task 3",
+            },
+            {
+              "title": "updated todo title",
+            }
+          ]
+        }
+      }
+    );
+    
+  })
+
+
+});
+
+const ADD_NEW_TODO = gql`
```

Comment:
Always try to help the reader by moving the graphql queries to the top of the file. Or even better: Put them inside the relevant `describe` block (if you nest your `describe` blocks according to your queries/mutations).